velox/functions/prestosql/types/JsonType.cpp

```diff
@@ -547,6 +547,8 @@ simdjson::error_code appendMapKey(
   using T = typename TypeTraits<kind>::NativeType;
   if constexpr (std::is_same_v<T, void>) {
     return simdjson::INCORRECT_TYPE;
+  } else if constexpr (std::is_same_v<T, UnknownValue>) {
+    return simdjson::EMPTY;
```

Comment:
`INCORRECT_TYPE` is a better error here

Reply:
Also it may be more readable if we add a specialization of `appendMapKey<TypeKind::UNKNOWN>` instead of `if` here

Reply:
Updated.